python/paddle/fluid/dygraph/jit.py

```diff
@@ -525,6 +528,58 @@ def _build_load_path_and_config(path, config):
     return model_path, config
 
 
+_save_pre_hooks_lock = threading.Lock()
+_save_pre_hooks = []
+
+
+class HookRemoveHelper(object):
+    """ A HookRemoveHelper that can be used to remove hook. """
+
+    def __init__(self, hook):
+        self._hook = hook
+
+    def remove(self):
+        remove_save_pre_hook(self._hook)
+
+
+def register_save_pre_hook(hook):
+    global _save_pre_hooks_lock
+    global _save_pre_hooks
+    _save_pre_hooks_lock.acquire()
+    if hook not in _save_pre_hooks:
+        _save_pre_hooks.append(hook)
+    _save_pre_hooks_lock.release()
+    return HookRemoveHelper(hook)
+
+
+def remove_save_pre_hook(hook):
+    global _save_pre_hooks_lock
+    global _save_pre_hooks
+    _save_pre_hooks_lock.acquire()
+    if hook in _save_pre_hooks:
+        _save_pre_hooks.remove(hook)
+    _save_pre_hooks_lock.release()
+
+
+def clear_save_pre_hooks():
```

Comment:
`clear_save_pre_hooks` used in what cases? Is `HookRemoveHelper.remove` enough?

Reply:
Currently not, but this is a convenient API to let user clear all hooks in one call.

Reply:
If the specific usage scenario is not clear, we recommended to use it as an internal method first, and then upgrade it to a public API if necessary in the future

Reply:
Done, currently rename `clear_pre_save_hooks`  with prefix `_` and make it internal use only.